pkg/controller/windowsmachine/metrics/metrics.go

```diff
@@ -0,0 +1,196 @@
+			return err
+		}
+		return nil
+	}
+
+	if addIPAddress {
+		return pc.addEndpointAddress()
+	}
+	return pc.removeEndpointAddress()
+}
+
+// isNodeInEndpoints checks if the given node is present in the list of addresses in the endpoints object
+func (pc *prometheusNodeConfig) isNodeInEndpoints() (string, error) {
+	endpoints, err := pc.k8sclientset.CoreV1().Endpoints(windowsMetricsEndpointsNamespace).Get(context.TODO(),
+		windowsMetricsEndpoints, metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrapf(err, "error getting endpoints object %v", windowsMetricsEndpoints)
+	}
+
+	foundNode := false
+	nodePosition := -1
+	for index, endpointAddress := range endpoints.Subsets[0].Addresses {
+		if pc.hostname == endpointAddress.TargetRef.Name {
+			foundNode = true
+			nodePosition = index
+			break
+		}
+	}
+	if !foundNode {
+		return "", errors.Wrapf(nil, "target node %s not found in Endpoints object %s",
+			pc.hostname, endpoints.Name)
+	}
+
+	return strconv.Itoa(nodePosition), nil
+}
+
+// addEndpointAddress adds node ip address to the list of subsets in Endpoints object
+func (pc *prometheusNodeConfig) addEndpointAddress() error {
+	_, err := pc.isNodeInEndpoints()
+	if err != nil {
+		// Add node if not present in the endpoints list
+		patchData := fmt.Sprintf("[{\"op\": \"add\", \"path\": \"/subsets/0/addresses/-\", \"value\":"+
+			" {\"ip\": \"%s\", \"targetRef\": {\"kind\": \"Node\", \"name\":\"%s\"}}}]",
+			pc.endpointAddress.IP, pc.hostname)
+
+		_, err := pc.k8sclientset.CoreV1().Endpoints(windowsMetricsEndpointsNamespace).Patch(context.TODO(),
+			windowsMetricsEndpoints, types.JSONPatchType, []byte(patchData), metav1.PatchOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "could not add node %v to endpoints object, %v", pc.hostname, pc.endpointAddress.IP)
+		}
+		return nil
+	}
+	log.Info("address added to endpoints object", "IPAddress", pc.endpointAddress.IP)
+	return nil
+}
+
+// removeEndpointAddress removes node ip address to the list of subsets in Endpoints object
+func (pc *prometheusNodeConfig) removeEndpointAddress() error {
+	nodeIPPosition, err := pc.isNodeInEndpoints()
+	// If node not found in the endpoints list, return without proceeding further
+	if err != nil {
+		return nil
+	}
+	patchData := fmt.Sprintf("[{\"op\": \"remove\", \"path\": \"/subsets/0/addresses/%s\"}]", nodeIPPosition)
+
+	_, err = pc.k8sclientset.CoreV1().Endpoints(windowsMetricsEndpointsNamespace).Patch(context.TODO(),
+		windowsMetricsEndpoints, types.JSONPatchType, []byte(patchData), metav1.PatchOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "could not remove node %v from endpoints object", pc.hostname)
+	}
+	log.Info("address removed from endpoints object", "IPAddress", pc.endpointAddress.IP)
+	return nil
+}
+
+// addMetrics will create the Services and Service Monitors to allow the operator export the metrics by using
+// the Prometheus operator
+func AddMetrics(ctx context.Context, cfg *rest.Config, namespace string) {
```

Comment:
This will now become just `Add()` so that there is not stuttering in the call sites: `metrics.Add()`.